default/scripting/ship_hulls/asteroid/SH_ASTEROID.focs.txt

```diff
@@ -29,6 +29,13 @@ Hull
                 OwnedBy empire = Source.Owner
             ]
         ]
+        ResourceSupplyConnected empire = Source.Owner condition = And [
+             Building name = "BLD_SHIPYARD_AST"
+             Or [
+                 OwnedBy empire = Source.Owner
+                 OwnedBy affiliation = AllyOf empire = Source.Owner
+             ]
+        ]
```

Comment:
To quote @LGM-Doyle:

> I used the same fixed for the asteroid armor and hulls. However, I don't know if it matter with the current implementation, since the asteroid processor must be in the same system. I tested it and it didn't make a difference. However, if the in system requirement changes, then the fix is already in.

Obviously the rationale is that in case the "in system" condition gets changed/removed (although I don't see that happening) a fix to at least still require those buildings to be within supply will be already in place.

I don't have a problem with that, it's just pointless IMO. Because if someone would make such changes, I'd expect them to put something else in place.

However, tweaking this additional condition so that you can't continue to build asteroid hulls in blockaded systems actually makes a lot of sense (after all, the rationale is that these asteroid processing facilites "harvest" and prepare asteroids, then these prepared things get transported to the shipyard, where ships are made out of them - a blockade should disrupt that).

The question is if the AI needs to be adjusted to this change - @freeorion/ai-team?